### PR TITLE
feat(oidc): Akamai Hosted Login

### PR DESCRIPTION
**Pull request by VancheeZze**
## Changes
- New `@bodiless/oidc` package.

## Test Instructions
- Try to Log In and Log Out using the button on the `/oidc` page.
- You can use `bob/bob` or `alice/alice` for login/password

## Related Issues
Closes: https://github.com/johnsonandjohnson/Bodiless-JS/issues/1147

## Notes
- Currently uses https://demo.identityserver.io/ as an Identity Provider.

### CODE REVIEW

**Comment by wodenx** [comment]
per discussions i think this can be constructed so as to be agnostic as to the OIDC provider, so i think we should rename to bodiless/oidc.  cc @marcopagliarulo  @VancheeZze

File: tsconfig.settings.json [hunk 1]
```diff
@@ -47,6 +47,8 @@
       "@bodiless/layouts-ui/*": ["bodiless-layouts-ui/src/*"],
       "@bodiless/navigation": ["bodiless-navigation/src"],
       "@bodiless/navigation/*": ["bodiless-navigation/src/*"],
+      "@bodiless/oicd": ["bodiless-oicd/src"],
```
Comment (awm086) [inline tsconfig.settings.json:50]: oidc?

File: packages/bodiless-oidc/src/AuthContext.tsx [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+ * @param args OIDC Sign In arguments
+ * @see SigninRedirectArgs
+ */
+export const withSignInOnClick = (
+  args?: SigninRedirectArgs,
+) => (Component: ComponentType) => (props: any) => {
+  // Property 'signIn' does not exist on type 'AuthContextProps | undefined'
+  // @ts-ignore
+  const { signIn } = useBodilessOidc();
+  return <Component {...props} onClick={() => signIn(args)} />;
```
Comment (marcopagliarulo) [inline packages/bodiless-oidc/src/AuthContext.tsx:46]: What do you think to use a conditional return in base of signIn null? Same applied below

**Comment by VancheeZze** [comment]
Closing this PR in favor of https://github.com/johnsonandjohnson/Bodiless-JS/pull/1190 created to the "main" branch.

File: packages/bodiless-oidc/src/AuthContext.tsx (hunk 2 above)
Comment (VancheeZze) [reply packages/bodiless-oidc/src/AuthContext.tsx:44]: `signIn` and `signOut` will always be there in this case. Those come with the provider and are always present. Added a type cast `as AuthContextProps`

**Pull request closed without merging**